### ISSUE TITLE
Thorium Go back in history failing with NVDA

### ISSUE THREAD

**Issue by GeorgeKerscher** [nvaccess/nvda#18654]
### Detailed description of the issue

In the Transition to Accessible EPUB working group in the DAISY Consortium, we are trying to add functionality to EPUB reading systems. We are interested in a go-back in history function. This is similar to following a link in a web page and then using a back key to return to that location.

In a EPUB reading a digital book context, the use cases for a go-back in history is when you follow a link to a footnote and want to go back, following a link to a glossary or bibliography, doing a search and want to return to your original reading positions, etc.

We have created an experimental EPUB title to test this functionality.

Developers at EDR lab who produce Thorium are working hard to implement this function with the CTRL + Backspace key. It is working with Jaws and  when no AT is used.

It is failing with NVDA in that you are not taken back to the exact position. The first four tests in the experimental test book fails; you are taken back to the heading in the chapter, but not to the anchor. We want to have this working perfectly.

It is a strange behavior, because other links in the book take you to the exact location.

### Why are the other templates not appropriate in this case?

This is not a feature request; it is not a bug report of NVDA; it seems not to be a Thorium problem. It seems to be some behavior we cannot figure out. We have really, really, really tried hard in the Thorium code to resolve this problem.
### Have you asked for advice on how to report this issue via a community 

We are using erly releases of Thorium and I have confirmed this behavior with Noelianrm1977@gmail.com

### Steps to reproduce or illustrate the issue (if applicable)

Download the experimental title:
[Reading History Navigation -Experimental](https://github.com/daisy/epub-accessibility-tests/releases/tag/reading-)history-navigation-experimental-1.0.0)

The beta version of Thorium that we have tested with and which works with Jaws is at:
[Nightly build of Thorium](https://github.com/edrlab/thorium-reader/releases/download/latest-windows-intel/[Thorium.Setup.3.3.0-alpha.1.16778323541.exe](https://github.com/edrlab/thorium-reader/releases/download/latest-windows-intel/Thorium.Setup.3.3.0-alpha.1.16778323541.exe))

### Expected outcome or behaviour (if applicable)

Open the EPUB in that version of Thorium. Go to the section with the tests. 
The section walks you through the tests with the expected outcomes. The first four tests are the ones that are failing. The footnote test in the same chapter, the footnote testin a endnotes section, the glossary reference and the bibliography reference.

Thorium has a setting to use pop-up footnotes. We think that it is best for screen reader users to disable this, because it would provide a consistent interface to go back. If this is enabled, escape should return to that location, but I suggest you disable this. The glossary and bibliograph reference is not a pop-up, and you will find that with NVDA, it does not return to the exact location, but to the top of the chapter.

### System configuration or software environment (if applicable)

NVDA 2025.2rc1 

### Additional information (if applicable)

The developer at EDR Lab working on Thorium is:
Daniel Weck
daniel.weck@gmail.com

Noelia and I George Kerscher
kerscher@montana.com

are happy to help.

**Comment by davidcarlisle** [comment]
link to the test file appears broken, intended link is (I think)

https://github.com/daisy/epub-accessibility-tests/releases/download/reading-history-navigation-experimental-1.0.0/Reading-History-Navigation-Experimental-v1.0.0.epub

**Comment by GeorgeKerscher** [comment]
The link you provided is correct. The original link is to the page about the commit and the link to the title is on that page.

**Comment by davidcarlisle** [comment]
@GeorgeKerscher  I added the link comment as there is a markdown error so the link in the issue description is 404 (but I am just a passer by I don't have edit access to correct that text)

**Comment by GeorgeKerscher** [comment]
I believe I have corrected the link to the EPUB title for testing.

**Comment by nvdaes** [comment]
cc: @michaelDCurran as a person who has worked in native selection mode, and @jcsteh as someone who works in Firefox, and perhaps can provide feedback about browse mode in Thorium.

**Comment by jcsteh** [comment]
Could you please provide technical details about how you are attempting to implement this in Thorium? That would be a lot simpler than needing to test it and reverse engineer what is being done.

Ideally, you would do this using URL fragments; i.e. `#someAnchor` after the URL, where someAnchor is the id of a section on the page you want to jump to. This is how same page links work. This should cause the page to scroll visually and NVDA should also pick this up and move the cursor to the targeted section.

**Comment by nvdaes** [comment]
@danielweck, can you please reply to @jcsteh ? I thik that your username was not mentioned here, so I pin you.

**Comment by danielweck** [comment]
Hello George, Noelia and NVDA team!
Before we move further let's assess the situation with Electron v37 which introduced a regression bug related to screen reader detection which could be the cause of the failures observed in Thorium (reminder: Electron ships with a particular revision of Chromium, the web browser engine that powers not only Thorium's user interface but also the webviews that render the HTML content documents of EPUB publications). In my own MacOS VoiceOver tests and in your Windows JAWS tests there seems to be no problems but because this is an edge case there is a possibility that so far only NVDA encountered it due to how the HTML document was scrolling at the time of test. I am preparing a Electron v36 test build of Thorium in order to eliminate the possibility that Electron might be the cause of failure.

**Comment by danielweck** [comment]
This is the Thorium issue / Electron v37 regression bug I was referring to:
https://github.com/edrlab/thorium-reader/issues/3106#issuecomment-3176531137

**Comment by danielweck** [comment]
@GeorgeKerscher @nvdaes Thorium is now building with a rollback to Electron v36, so we can run NVDA tests.

Build status:
https://github.com/edrlab/thorium-reader/actions/runs/16904374456

Windows Intel x64 builds:
https://github.com/edrlab/thorium-reader/releases/tag/latest-windows-intel

**Comment by jcsteh** [comment]
Ah, since Thorium uses Chromium (not Mozilla Gecko), I likely can't help here.

**Comment by danielweck** [comment]
I believe this is fixed in the latest Thorium release version 3.2.2 thanks to a revision of Electron version 37 which includes a bugfix for a regression that prevented detection of screen readers.
https://github.com/edrlab/thorium-reader/releases/tag/v3.2.2

**Comment by seanbudd** [comment]
Thanks, closing as fixed. @GeorgeKerscher please let us know if you are still having issues with the latest version

**Comment by GeorgeKerscher** [comment]
I want to thank Daniel who worked so hard to get this fixed!
